src/core/ColorUtil.cc

```diff
@@ -100,4 +103,194 @@ std::optional<Color4f> parse_hex_color(const std::string& hex) {
   return rgba;
 }
 
-} // namespace
\ No newline at end of file
+std::optional<Color4f> parse_web_color(const std::string& col) {
+  std::string colorname = boost::algorithm::to_lower_copy(col);
```

Comment:
The reason for test failures is that previously, the results of to_lower() was used to also print errors/warnings. I think your approach is more correct, you just have to update tests accordingly